guide_drafts/model-derives.md

```diff
@@ -0,0 +1,769 @@
+
+Throughout this guide, we will be looking at examples for each trait and how they interact together. Some of the code will be implementing basic CRUD operations, however the details of those operations will be not be covered beyond their relevance to the demonstrated trait.
+
+- [Queryable](#queryable)
+- [Insertable](#insertable)
+- [Identifiable](#identifiable)
+- [AsChangeset](#aschangeset)
+- [Associations](#associations)
+
+## <a name="queryable"></a>Queryable
+
+Using the [#[derive(Queryable)]](http://docs.diesel.rs/diesel/query_source/trait.Queryable.html) trait annotation on your model struct allows records to be queried from the database. This means that any time you are using the `load()`, `get_result()`, `get_results()`, and `first()` methods to execute your queries, you must have `Queryable` implemented on your model.
+
+There are some considerations to be made when implementing the `Queryable` trait. You must include all the table's fields in your model struct. The implementation details for this rule are only important if you wish to `impl` the trait by hand and will be covered later in this section.  Your struct name should be the singular of its corresponding table name. Queryable by itself does not support the `#[table_name=""]` attribute. 
+
+### Example 
+
+*Note: src/lib.rs with #[macro_use] extern crate diesel_codegen; not shown*
+
+*src/models.rs*
+```rust
+#[derive(Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+```
+
+*src/main.rs*
+```rust
+// externs, database connection, and using statement code omitted
+
+fn main() {
+    // The following will return all Users as a QueryResult<Vec<User>>
+    let query_result = users.load::<User>(&db_connection)
+      .expect("Error loading users");
+}
+```
+
+When deriving Queryable for a model, the struct must have all fields defined. Queryable's associated type, Row,  implements `FromSqlRow`, which provides the `fields_needed()` method. That method returns the number of fields to be consumed by the row. If we were to comment out all the three `String` fields on our User struct, we would see the following error.
+
+```rust
+error[E0277]: the trait bound `(i32,): diesel::types::FromSqlRow<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` is not satisfied
+  --> src/bin/main.rs:21:28
+   |
+21 |     let query_result = users.load::<User>(&db_connection).expect("Error loading users");
+   |                            ^^^^ the trait `diesel::types::FromSqlRow<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` is not implemented for `(i32,)`
```

Comment:
I'll look into the formatting for this. It is under `.load`. Looks fine in vim && my markdown preview, but there might be some inconsistent spacing issues from when I pasted it from the cli.